### DISCUSSION TITLE
Welcome to seispy Discussions!

### DISCUSSION THREAD

**Discussion by xumi1993** [xumi1993/seispy#20] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by tianjueli** [comment]
Hi Mijian, 
I'm using python-seispy-1.2.16 to do receiver function analysis. Now I received the warning info when **stel** (station elevation) is negative as shown bellow:
_"2022-04-06 19:01:04,435 [RF2depth] INFO: the 1th/1 station with 1162 events
/opt/anaconda3/lib/python3.8/site-packages/scipy/interpolate/interpolate.py:684: RuntimeWarning: invalid value encountered in greater
  above_bounds = x_new > self.x[-1]"_
Though there's the warning info, the time-to-depth conversion still looks correct.

However, wrong time-to-depth conversion occurs when **stel** is large than 4000 though there's no any warning or error info occurred:
![Screenshot from 2022-04-06 19-21-36](https://user-images.githubusercontent.com/38813823/161963844-dda77803-4bc6-47a3-a2dc-24a5f83ef55c.png)

Correct time-to-depth conversion when I tried **stel** = 0, 3000 and -5000:
![Screenshot from 2022-04-06 19-22-59](https://user-images.githubusercontent.com/38813823/161964235-ca90c586-a60f-40cd-9689-a1c778e2dd27.png)

**Reply by xumi1993** [reply]
I've checked PRFs in the station `CB.NJ2`. It seems good.

```python
from seispy.rfcorrect import RFStation
import matplotlib.pyplot as plt
import numpy as np
```


```python
rfsta = RFStation('RFresult/CB.NJ2', prime_comp='Q')
```


```python
rfsta.stel = 0
rf_e = rfsta.psrf2depth()
rfsta.stel = 4000
rf_4000 = rfsta.psrf2depth()
rfsta.stel = -4000
rf_m4000 = rfsta.psrf2depth()
rfsta.stel = 7000
rf_7000 = rfsta.psrf2depth()
```

## For single PRF
```python
plt.style.use('bmh')
plt.figure(figsize=(10, 3))
plt.plot(np.arange(150), rf_e[0,:], label='0')
plt.plot(np.arange(150), rf_4000[0,:], label='4000')
plt.plot(np.arange(150), rf_m4000[0,:], label='-4000')
plt.plot(np.arange(150), rf_7000[0,:], label='7000')
plt.legend()

```
![output_3_1](https://user-images.githubusercontent.com/7437523/161979869-0ba8e0fb-97d3-4db3-86ef-4e18180848f3.png)


## For average RFs 
```python
plt.style.use('bmh')
plt.figure(figsize=(10, 3))
plt.plot(np.arange(150), np.mean(rf_e, axis=0), label='0')
plt.plot(np.arange(150), np.mean(rf_4000, axis=0), label='4000')
plt.plot(np.arange(150), np.mean(rf_m4000, axis=0), label='-4000')
plt.plot(np.arange(150), np.mean(rf_7000, axis=0), label='7000')
plt.legend()

```

    
![output_4_1](https://user-images.githubusercontent.com/7437523/161979900-dc01c46f-3ae0-419c-89a2-799ab57d00c7.png)
